components/SearchBar.tsx

```diff
@@ -1,65 +1,119 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
+import { motion } from "framer-motion";
+import { AiOutlineClose, AiOutlineSearch } from 'react-icons/ai';
+import type { CourseDTO } from "@/types/CourseDTO";
 
 export interface SearchBarProps {
-  onSearchClick: (searchValue: string) => void;
+  onSearchClick: (searchValue: string, courseList: CourseDTO[]) => void;
 }
 
 const SearchBar = ({ onSearchClick }: SearchBarProps) => {
   const [searchValue, setSearchValue] = useState("");
+  const [text, setText] = useState("");
+  const [courses, setCourses] = useState<CourseDTO[]>([]);
+  const [list, setList] = useState<CourseDTO[]>([]);
 
   const handleSearchChange = (event: any) => {
-    setSearchValue(event.target.value);
+    const currentValue = event.target.value
+    setSearchValue(currentValue);
+    setText(currentValue);
+
+    const res = courses.filter((c) =>
+      c.course_code.toLowerCase().includes(currentValue.toLowerCase())
+    ).slice(0, 12);
+    setList([...res]);
+  };
+
+  const onEnter = (searchValue: string) => {
+    setText("");
+    setSearchValue(searchValue);
+    onSearchClick(searchValue, list);
+  };
+
+  const onSelect = (searchSuggestedValue: string) => {
+    setText("");
+    setSearchValue(searchSuggestedValue);
+    onSearchClick(searchSuggestedValue, list);
+  };
+
+  const onSelectClear = (searchValue: string) => {
+    setSearchValue(searchValue);
+    setText("");
   };
 
+  // Getting course data from database
+  useEffect(() => {
+    fetch(
+      "/api/courses?",
+      {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      }
+    ).then((response) => {
+      if (response.ok) {
+        response.json().then((data: CourseDTO[]) => {
```

Comment:
A little nitpicky, but we can remove the braces around this anonymous function, right? It’s just a single statement.